src/components/Common/AvatarEditModal.tsx

```diff
@@ -78,51 +81,73 @@ const AvatarEditModal = ({
     VideoConstraints.user,
   );
   const { t } = useTranslation();
+  const { captureWebcamImage, processCroppedImage } = useImageCapture();
   const [isDragging, setIsDragging] = useState(false);
   const { requestPermission } = useMediaDevicePermission();
+  const [isCropping, setIsCropping] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
   const handleSwitchCamera = useCallback(() => {
     setConstraint(
       constraint.facingMode === "user"
         ? VideoConstraints.environment
         : VideoConstraints.user,
     );
-  }, []);
+  }, [constraint]);
+
+  const captureImage = async () => {
+    const { screenshot, file, error } = await captureWebcamImage(webRef);
 
-  const captureImage = () => {
-    if (webRef.current) {
-      setPreviewImage(webRef.current.getScreenshot());
+    if (error) {
+      return;
+    }
+
+    if (screenshot && file) {
+      setPreviewImage(screenshot);
+      setSelectedFile(file);
+      // Enter cropping mode when capturing a new image
+      setIsCropping(true);
     }
-    const canvas = webRef.current?.getCanvas();
-    canvas?.toBlob((blob) => {
-      if (blob) {
-        const myFile = new File([blob], "image.png", {
-          type: blob.type,
-        });
-        setSelectedFile(myFile);
-      } else {
-        toast.error(t("failed_to_capture_image"));
-      }
-    });
   };
-  const stopCamera = useCallback(() => {
-    try {
-      if (webRef.current) {
-        const openCamera = webRef.current?.video?.srcObject as MediaStream;
-        if (openCamera) {
-          openCamera.getTracks().forEach((track) => track.stop());
+
+  const handleCroppedImageComplete = useCallback(
+    async (croppedImage: string) => {
+      setIsProcessing(true);
+
+      try {
+        const { file, error } = await processCroppedImage(croppedImage);
+
+        if (error) {
+          setErrorMessage(error);
+          return;
+        }
+
+        if (file) {
+          setSelectedFile(file);
+          setPreview(croppedImage);
+          setIsCropping(false);
+          setIsCaptureImgBeingUploaded(false);
+          setIsProcessing(false);
+          setErrorMessage(null);
         }
+      } catch (error) {
+        console.error("Error processing cropped image:", error);
+        setErrorMessage(t("AVATAR_EDIT__ERROR_PROCESSING_IMAGE"));
+      } finally {
+        setIsProcessing(false);
       }
-    } catch {
-      toast.error("Failed to stop camera");
-    } finally {
-      setIsCameraOpen(false);
-    }
-  }, []);
+    },
+    [processCroppedImage, t],
```

Comment:
dependency shouldn't be necessary here either since both are functions imported from packages and unlikely to change.